gobblin-data-management/src/main/java/gobblin/data/management/copy/replication/ConfigBasedDatasetsFinder.java

```diff
@@ -167,14 +172,15 @@ public ConfigBasedDatasetsFinder(FileSystem fs, Properties jobProps) throws IOEx
     Set<URI> disabledURISet = new HashSet();
     if (this.blacklistURNs.isPresent()) {
       for(String urn : this.blacklistURNs.get()) {
-        disabledURISet.add(this.datasetURNtoURI(urn));
+        disabledURISet.addAll(this.datasetURNtoURI(urn));
       }
     }
 
     try {
       // get all the URIs which imports {@link #replicationTag} or all from whitelistURNs
       allDatasetURIs = this.whitelistURNs.isPresent()
-          ? this.whitelistURNs.get().stream().map(u -> this.datasetURNtoURI(u)).collect(Collectors.toList()) : configClient.getImportedBy(new URI(whitelistTag.toString()), true);
+          ? this.whitelistURNs.get().stream().flatMap(u -> this.datasetURNtoURI(u).stream()).collect(Collectors.toList())
+          : configClient.getImportedBy(new URI(whitelistTag.toString()), true);
       populateDisabledURIs(disabledURISet);
```

Comment:
The configuration options are not symmetric here, which can lead to confusion and does not allow all override possibilities. The code is ignoring the config store whitelist tag when a job-level whitelist is specified, but it is adding the job-level blacklist to the tag-based blacklist.